dotnet/src/Agents/Abstractions/Definition/AgentCreationOptions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Optional parameters for agent creation.
+/// </summary>
+public class AgentCreationOptions
+{
+    /// <summary>
+    /// Gets or sets the <see cref="Kernel"/>, a kernel instance to resolve services.
+    /// </summary>
+    public Kernel? Kernel { get; init; } = null;
+
+    /// <summary>
+    /// Gets or sets the <see cref="IPromptTemplateFactory"/>, a factory for prompt templates for one or more prompt template formats.
+    /// </summary>
+    public IPromptTemplateFactory? PromptTemplateFactory { get; init; } = null;
+
+    /// <summary>
+    /// Gets or sets the <see cref="IConfiguration"/>, a set of key/value agent configuration properties.
+    /// </summary>
+    public IConfiguration? Configuration { get; init; } = null;
```

Comment:
KernelArguments instead of `IConfiguration` ?

```suggestion
    public IConfiguration? Configuration { get; init; } = null;
```

Reply:
IN my opinion adding this `Configuration` new templating thing could be an extension not necesarily a native support. So since we are already using the `PromptTemplate` logic here, would make sense to just use our `KernelArguments` as the input for templating the agent declaration.

Reply:
The agent declarative specification allows for configuration values e.g. API Keys, connection strings, etc. to be inserted into the agent definition. For .Net using `IConfiguration` would be the idiomatic was to do this. This is different from templating the Agent instructions.